mongo/cursor.go

```diff
@@ -203,10 +216,59 @@ func (c *Cursor) next(ctx context.Context, nonBlocking bool) bool {
 	}
 }
 
+func getDecoder(
+	data []byte,
+	opts *options.BSONOptions,
+	reg *bsoncodec.Registry,
+) (*bson.Decoder, error) {
+	dec, err := bson.NewDecoder(bsonrw.NewBSONDocumentReader(data))
+	if err != nil {
+		return nil, err
+	}
+
+	if opts != nil {
+		if opts.AllowTruncatingDoubles {
+			dec.AllowTruncatingDoubles()
+		}
+		if opts.BinaryAsSlice {
+			dec.BinaryAsSlice()
+		}
+		if opts.DefaultDocumentD {
+			dec.DefaultDocumentD()
+		}
+		if opts.DefaultDocumentM {
+			dec.DefaultDocumentM()
+		}
+		if opts.UseJSONStructTags {
+			dec.UseJSONStructTags()
+		}
+		if opts.UseLocalTimeZone {
+			dec.UseLocalTimeZone()
+		}
+		if opts.ZeroMaps {
+			dec.ZeroMaps()
+		}
+		if opts.ZeroStructs {
+			dec.ZeroStructs()
+		}
+	}
+
+	if reg != nil {
+		_ = dec.SetRegistry(reg)
```

Comment:
Why don't we check this error?

Reply:
[SetRegistry](https://github.com/mongodb/mongo-go-driver/blob/e748329f8e9b349010d3f185592c76ee22719b63/bson/decoder.go#L145-L148) never returns an error, so it mostly seems unnecessary. If for some reason we ever decided `SetRegistry` needs to return an error, that would create an unhandled error condition, so it's definitely safer to include the error checking. Probably worth adding for safety.